js/hooks/useChannel.ts

```diff
@@ -0,0 +1,90 @@
+import { reload } from "../browser";
+import { join } from "../util/channel";
+import { usePageVisibility } from "./usePageVisibility";
+import { Channel, Socket } from "phoenix";
+import { useEffect, useState } from "react";
+import { z } from "zod";
+
+/**
+ * Maintain a Phoenix.Channel connection (disconnecting when the page is not visible).
+ * The returned channel has not joined yet, so callers must `.join().receive(...)` themselves.
+ * This function will handle leaving the channel when the effect is unmounted; callers should not call `.leave()` themselves.
+ * The channel will change its identity when the topic changes or when the page visibility changes (it will be null while the page is invisible).
+ * WARNING: If you join a channel which is already joined, the existing subscription will be silently replaced, which is not what you want.
+ */
+export const useRawChannel = ({
+  socket,
+  actuallyConnect = true,
+  joinParams,
+  topic,
+}: {
+  socket: Socket | null;
+  actuallyConnect?: boolean;
```

Comment:
I think YAGNI for `actuallyConnect`.

It has a few uses in Glides:
- Scenario mode
- Manage runs when looking at tomorrow (which doesn't have live data)
- The realtime map page (when you aren't showing some data on the map)
- The predictions channel, when the panel is closed.

You _might_ need it some day, but I don't think any of those situations will pop up, and it'll be easy enough to re-add if needed.